Sources/ErgastAPI/ErgastAPIService.swift

```diff
@@ -0,0 +1,71 @@
+//
+//  ErgastAPIService.swift
+//
+//
+//  Created by Giovanni Noa on 4/28/20.
+//
+
+import Foundation
+import UIKit
+
+// MARK: - Ergast API Service
+/// Ergast service for interacting with the Ergast endpoints within the ErgastAPI package.
+enum ErgastAPIService {
+    // MARK: - Static Networking Functions
+    /// Callback-based networking data task. Called by the internal fetch function.
+    /// - Parameters:
+    ///   - url: URL to reach.
+    ///   - session: URLSession instance.
+    ///   - completion: Asynchronous closure to inject functionality once the network interaction completes.
+    private static func dataTask(_ url: URL,
+                                 _ session: URLSession,
+                                 _ completion: @escaping ((Result<Data, ErgastAPIError>) -> Void))
+    {
+        session.dataTask(with: url) { data, response, error in
+            #warning("Handle the rest of networking")
+            guard let data = data, error == nil else {
+                completion(.failure(.network(error!.localizedDescription)))
```

Comment:
Avoid force unwrapping. Another approach is to flip it:

```swift
if let error = error, data == nil {
  completion(.failure(.network(error.localizedDescription)))
  return
}
```

Reply:
What do I do for the data part giving me a optional error? I have to unwrap it as well

Reply:
@gionoa maybe split it into two?